IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -597,9 +604,19 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
         {
             *error = MSIDCreateError(MSIDKeychainErrorDomain, status, @"Failed to remove items from keychain.", nil, nil, nil, context.correlationId, nil);
         }
-        MSID_LOG_ERROR(context, @"Failed to delete keychain items (status: %d)", (int)status);
-
-        return NO;
+        MSID_LOG_WARN(context, @"Failed to delete keychain items (status: %d), using fallback", (int)status);
+        // If for some reason the single SecItemDelete() fails, delete as many of the individual items from the keychain as possible.
```

Comment:
why is this necessary?

Reply:
On our CI build box I was seeing the SecItemDelete() fail with error -25244. To investigate, I added code to iterate the items to find more info about the item or items that were causing it to fail and found that the error was no longer happening. In any case, especially since this is code for testing, it seems like trying to cleanup the items we can rather than letting the whole delete fail would be useful. Alternatively, we could wait until we have another repro in the future and take this up again at that time.

Reply:
I think the question here is if this is behavior we want to have in production or it's just for testing. If it's just for testing, this code shouldn't be here. If it's for production, we should rationalize this partial delete.

Reply:
The whole clearWithContext method is just for testing, though. That said, I should wait on this part until we have another repro of the problem and debug it further at that time so we can be more confident in the change. I'll remove this part from the current PR.